src/MongoDB/Logging/Logger.stub.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * @generate-class-entries static
+ * @generate-function-entries
+ */
+
+namespace MongoDB\Driver\Logging;
+
+interface Logger
+{
+    /**
+     * @var int
+     * @cvalue MONGOC_LOG_LEVEL_ERROR
+     */
+    public const LEVEL_ERROR = UNKNOWN;
+
+    /**
+     * @var int
+     * @cvalue MONGOC_LOG_LEVEL_CRITICAL
+     */
+    public const LEVEL_CRITICAL = UNKNOWN;
+
+    /**
+     * @var int
+     * @cvalue MONGOC_LOG_LEVEL_WARNING
+     */
+    public const LEVEL_WARNING = UNKNOWN;
+
+    /**
+     * @var int
+     * @cvalue MONGOC_LOG_LEVEL_MESSAGE
+     */
+    public const LEVEL_MESSAGE = UNKNOWN;
+
+    /**
+     * @var int
+     * @cvalue MONGOC_LOG_LEVEL_INFO
+     */
+    public const LEVEL_INFO = UNKNOWN;
+
+    /**
+     * @var int
+     * @cvalue MONGOC_LOG_LEVEL_DEBUG
+     */
+    public const LEVEL_DEBUG = UNKNOWN;
+
+    /**
+     * @var int
+     * @cvalue MONGOC_LOG_LEVEL_TRACE
+     */
+    public const LEVEL_TRACE = UNKNOWN;
+
+    public function log(int $level, string $domain, string $message): void;
```

Comment:
I'm not entirely happy with this API, as I think most would expect the `log()` method of a `Logger` interface to be used to _write_ log messages, not receive them.

And then I considered renaming this class to `LogSubscriber` and found myself asking why it's in a separate namespace to begin with.

What do you think about the following:

 * Rename `MongoDB\Logging\Logger` to `MongoDB\Monitoring\LogSubscriber` (leave its `log()` method as-is).
 * `MongoDB\Monitoring\LogSubscriber` extends the Subscriber interface, like the APM and SDAM listeners.
 * Allow a log listener to be registered through the existing `addSubscriber()` methods. LogSubscriber would then get treated differently from the other two subscriber interfaces.
 * We could choose to prohibit passing a LogSubscriber to `Manager::addSubscriber()` and force users to add them with `MongoDB\Driver\Monitoring\addSubscriber()` because libmongoc only has a global logger. If that ever changes and we can implement client-level logging atop libmongoc, we can then allow `Manager::addSubscriber()` to take a LogSubscriber. Basically, this would avoid a potential BC issue.

Logging and monitoring are already intertwined (the APM spec was renamed to [Command Logging and Monitoring](https://github.com/mongodb/specifications/blob/master/source/command-logging-and-monitoring/command-logging-and-monitoring.rst), so I don't think this would be too strange a proposal. It would also let us avoid expanding our API/docs footprint.

Reply:
Since we have decided to not have a `$minLogLevel`, I think the last reason to not make it a `LogSubscriber` went out the door. IMO this sounds like a good plan.

Reply:
SGTM. I'll be very pleased with that API.

And we'll have `Manager::addSubscriber()` throw for now, instead of introducing an intermediary interface to distinguish global and Manager subscribers (see [Slack thread](https://mongodb.slack.com/archives/C026UA4GF0B/p1694619954582389?thread_ts=1694618364.328659&cid=C026UA4GF0B)).

Reply:
This is now implemented, and I also added tests for registering a single object that is both an APM _and_ log subscriber.